EIPS/eip-rpc.md

```diff
@@ -0,0 +1,2314 @@
+    }]
+}' <url>
+
+# Response
+{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "result": "0xe670ec64341771606e55d6b4ca35a1a6b75ee3d5145a99d05921026d1527331"
+}
+```
+---
+</details>
+
+<details>
+<summary><code><strong>eth_sign</strong></code></summary>
+
+#### Description
+
+Calculates an Ethereum-specific signature in the form of `keccak256("\x19Ethereum Signed Message:\n" + len(message) + message))`
+
+#### Parameters
+
+|#|Type|Description|
+|-|-|-|
+|1|{[`Data`](#data)}|address to use for signing|
+|2|{[`Data`](#data)}|data to sign|
+
+#### Returns
+
+{[`Data`](#data)} - signature hash of the provided data
+
+#### Example
+
+```sh
+# Request
+curl -X POST --data '{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "method": "eth_sign",
+    "params": ["0x9b2055d370f73ec7d8a03e965129118dc8f5bf83", "0xdeadbeaf"]
+}' <url>
+
+# Response
+{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "result": "0xa3f20717a250c2b0b729b7e5becbff67fdaef7e0699da4de7ca5895b02a170a12d887fd3b17bfdce3481f10bea41f45ba9f709d39ce8325427b57afcfc994cee1b"
+}
+```
+---
+</details>
+
+<details>
+<summary><code><strong>eth_signTransaction</strong></code></summary>
+
+#### Description
+
+Signs a transaction that can be submitted to the network at a later time using with `eth_sendRawTransaction`
+
+#### Parameters
+
+|#|Type|Description|
+|-|-|-|
+|1|{`object`}|@property {[`Data`](#data)} `from` - transaction sender<br/>@property {[`Data`](#data)} `[to]` - transaction recipient<br/>@property {[`Quantity`](#quantity)} `[gas="0x15f90"]` - gas provided for transaction execution<br/>@property {[`Quantity`](#quantity)} `[gasPrice]` - price in wei of each gas used<br/>@property {[`Quantity`](#quantity)} `[value]` - value in wei sent with this transaction<br/>@property {[`Data`](#data)} `[data]` - contract code or a hashed method call with encoded args<br/>@property {[`Quantity`](#quantity)} `[nonce]` - unique number identifying this transaction|
+
+#### Returns
+
+{[`Data`](#data)} - signature hash of the transaction object
+
+#### Example
+
+```sh
+# Request
+curl -X POST --data '{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "method": "eth_signTransaction",
+    "params": [{
+        "data": "0xd46e8dd67c5d32be8d46e8dd67c5d32be8058bb8eb970870f072445675058bb8eb970870f072445675",
+        "from": "0xb60e8dd61c5d32be8058bb8eb970870f07233155",
+        "gas": "0x76c0",
+        "gasPrice": "0x9184e72a000",
+        "to": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+        "value": "0x9184e72a"
+    }]
+}' <url>
+
+# Response
+{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "result": "0xa3f20717a250c2b0b729b7e5becbff67fdaef7e0699da4de7ca5895b02a170a12d887fd3b17bfdce3481f10bea41f45ba9f709d39ce8325427b57afcfc994cee1b"
+}
+```
+---
+</details>
+
+<details>
+<summary><code><strong>eth_signTypedData</strong></code></summary>
+
+#### Description
+
+Calculates an Ethereum-specific signature in the form of `keccak256("\x19Ethereum Signed Message:\n" + len(message) + message))`
```

Comment:
This description looks incorrect (probably copy/pasted from `eth_sign`).  This should be updated to reference EIP-712

Reply:
Based on the RPC method [defined in EIP-712](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md#eth_signtypeddata), `eth_signTypedData` still calculates a hash using `keccak256("\x19Ethereum Signed Message:\n" + len(message) + message))`. As I understand it, the only difference between this method and `eth_sign` is the `message` itself.

Reply:
@pipermerriam @bitpshr I know this conversation happened a while ago, but I came across this while pulling EIP-712 in to web3.py and wanted some feedback. The way I am reading EIP-712 is that `eth_signTypedData` does not necessarily append `\x19Ethereum Signed Message:\n`, whereas `eth_sign` does. I'm looking at [this part](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md#specification) of the specification. I wonder if the link to the `eth_signTypedData` spec above is in fact a copy paste error, because it says the same thing verbatim in the [eth_sign RPC docs](https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_sign). 

Also, let me know if there is a better place for this discussion.

Reply:
@kclowes based on my recolection of the spec, I believe you are correct.  the `\x19Ethereum Signed Message` is indeed not part of either of the new signing EIPs.